main/src/library/String.flix

```diff
@@ -65,9 +65,9 @@ namespace String {
     ///
     /// Splits the string `s` around matches of the regular expression `regex`.
     ///
-    pub def split(regex: {regex :: String}, s: String): List[String] & Impure =
-        import java.lang.String.split(String): Array[String] & Impure;
-        Array.toList(split(s, regex.regex))
+    pub def split(regex: {regex :: String}, s: String): List[String] =
+        import java.lang.String.split(String): Array[String] & Pure;
+        Array.toList(split(s, regex.regex)) as & Pure
```

Comment:
Is it a problem, that `Array.toList` is `Impure`?

Reply:
This is an okay cast-- at least until regions arrive.